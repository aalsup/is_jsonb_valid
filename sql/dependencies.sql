-- dependencies
-- neither
SELECT is_jsonb_valid('{"dependencies":{"bar":["foo"]}}', '{}');
-- nondependant
SELECT is_jsonb_valid('{"dependencies":{"bar":["foo"]}}', '{"foo":1}');
-- with dependency
SELECT is_jsonb_valid('{"dependencies":{"bar":["foo"]}}', '{"foo":1,"bar":2}');
-- missing dependency
SELECT is_jsonb_valid('{"dependencies":{"bar":["foo"]}}', '{"bar":2}');
-- ignores arrays
SELECT is_jsonb_valid('{"dependencies":{"bar":["foo"]}}', '["bar"]');
-- ignores strings
SELECT is_jsonb_valid('{"dependencies":{"bar":["foo"]}}', '"foobar"');
-- ignores other non-objects
SELECT is_jsonb_valid('{"dependencies":{"bar":["foo"]}}', '12');
-- multiple dependencies
-- neither
SELECT is_jsonb_valid('{"dependencies":{"quux":["foo","bar"]}}', '{}');
-- nondependants
SELECT is_jsonb_valid('{"dependencies":{"quux":["foo","bar"]}}', '{"foo":1,"bar":2}');
-- with dependencies
SELECT is_jsonb_valid('{"dependencies":{"quux":["foo","bar"]}}', '{"foo":1,"bar":2,"quux":3}');
-- missing dependency
SELECT is_jsonb_valid('{"dependencies":{"quux":["foo","bar"]}}', '{"foo":1,"quux":2}');
-- missing other dependency
SELECT is_jsonb_valid('{"dependencies":{"quux":["foo","bar"]}}', '{"bar":1,"quux":2}');
-- missing both dependencies
SELECT is_jsonb_valid('{"dependencies":{"quux":["foo","bar"]}}', '{"quux":1}');
-- multiple dependencies subschema
-- valid
SELECT is_jsonb_valid('{"dependencies":{"bar":{"properties":{"foo":{"type":"integer"},"bar":{"type":"integer"}}}}}', '{"foo":1,"bar":2}');
-- no dependency
SELECT is_jsonb_valid('{"dependencies":{"bar":{"properties":{"foo":{"type":"integer"},"bar":{"type":"integer"}}}}}', '{"foo":"quux"}');
-- wrong type
SELECT is_jsonb_valid('{"dependencies":{"bar":{"properties":{"foo":{"type":"integer"},"bar":{"type":"integer"}}}}}', '{"foo":"quux","bar":2}');
-- wrong type other
SELECT is_jsonb_valid('{"dependencies":{"bar":{"properties":{"foo":{"type":"integer"},"bar":{"type":"integer"}}}}}', '{"foo":2,"bar":"quux"}');
-- wrong type both
SELECT is_jsonb_valid('{"dependencies":{"bar":{"properties":{"foo":{"type":"integer"},"bar":{"type":"integer"}}}}}', '{"foo":"quux","bar":"quux"}');