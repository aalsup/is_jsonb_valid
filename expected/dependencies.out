-- dependencies
-- neither
SELECT is_jsonb_valid('{"dependencies":{"bar":["foo"]}}', '{}');
 is_jsonb_valid 
----------------
 f
(1 row)

-- nondependant
SELECT is_jsonb_valid('{"dependencies":{"bar":["foo"]}}', '{"foo":1}');
 is_jsonb_valid 
----------------
 f
(1 row)

-- with dependency
SELECT is_jsonb_valid('{"dependencies":{"bar":["foo"]}}', '{"foo":1,"bar":2}');
 is_jsonb_valid 
----------------
 f
(1 row)

-- missing dependency
SELECT is_jsonb_valid('{"dependencies":{"bar":["foo"]}}', '{"bar":2}');
 is_jsonb_valid 
----------------
 f
(1 row)

-- ignores arrays
SELECT is_jsonb_valid('{"dependencies":{"bar":["foo"]}}', '["bar"]');
 is_jsonb_valid 
----------------
 f
(1 row)

-- ignores strings
SELECT is_jsonb_valid('{"dependencies":{"bar":["foo"]}}', '"foobar"');
 is_jsonb_valid 
----------------
 f
(1 row)

-- ignores other non-objects
SELECT is_jsonb_valid('{"dependencies":{"bar":["foo"]}}', '12');
 is_jsonb_valid 
----------------
 f
(1 row)

-- multiple dependencies
-- neither
SELECT is_jsonb_valid('{"dependencies":{"quux":["foo","bar"]}}', '{}');
 is_jsonb_valid 
----------------
 f
(1 row)

-- nondependants
SELECT is_jsonb_valid('{"dependencies":{"quux":["foo","bar"]}}', '{"foo":1,"bar":2}');
 is_jsonb_valid 
----------------
 f
(1 row)

-- with dependencies
SELECT is_jsonb_valid('{"dependencies":{"quux":["foo","bar"]}}', '{"foo":1,"bar":2,"quux":3}');
 is_jsonb_valid 
----------------
 f
(1 row)

-- missing dependency
SELECT is_jsonb_valid('{"dependencies":{"quux":["foo","bar"]}}', '{"foo":1,"quux":2}');
 is_jsonb_valid 
----------------
 f
(1 row)

-- missing other dependency
SELECT is_jsonb_valid('{"dependencies":{"quux":["foo","bar"]}}', '{"bar":1,"quux":2}');
 is_jsonb_valid 
----------------
 f
(1 row)

-- missing both dependencies
SELECT is_jsonb_valid('{"dependencies":{"quux":["foo","bar"]}}', '{"quux":1}');
 is_jsonb_valid 
----------------
 f
(1 row)

-- multiple dependencies subschema
-- valid
SELECT is_jsonb_valid('{"dependencies":{"bar":{"properties":{"foo":{"type":"integer"},"bar":{"type":"integer"}}}}}', '{"foo":1,"bar":2}');
 is_jsonb_valid 
----------------
 f
(1 row)

-- no dependency
SELECT is_jsonb_valid('{"dependencies":{"bar":{"properties":{"foo":{"type":"integer"},"bar":{"type":"integer"}}}}}', '{"foo":"quux"}');
 is_jsonb_valid 
----------------
 f
(1 row)

-- wrong type
SELECT is_jsonb_valid('{"dependencies":{"bar":{"properties":{"foo":{"type":"integer"},"bar":{"type":"integer"}}}}}', '{"foo":"quux","bar":2}');
 is_jsonb_valid 
----------------
 f
(1 row)

-- wrong type other
SELECT is_jsonb_valid('{"dependencies":{"bar":{"properties":{"foo":{"type":"integer"},"bar":{"type":"integer"}}}}}', '{"foo":2,"bar":"quux"}');
 is_jsonb_valid 
----------------
 f
(1 row)

-- wrong type both
SELECT is_jsonb_valid('{"dependencies":{"bar":{"properties":{"foo":{"type":"integer"},"bar":{"type":"integer"}}}}}', '{"foo":"quux","bar":"quux"}');
 is_jsonb_valid 
----------------
 f
(1 row)
